src/pymor/models/transforms.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+from scipy.linalg import null_space
+
+from pymor.core.base import ImmutableObject
+
+
+class MoebiusTransformation(ImmutableObject):
+    r"""Maps the Riemann sphere onto itself.
+
+    A Moebius transformation
+
+    .. math::
+        M(s) = \frac{as+b}{cs+b}
+
+    is determined by the coefficients :math:`a,b,c,d\in\mathbb{C}`. The Moebius transformations form
+    a group under composition, therefore the `__matmul__` operator is defined to yield a
+    |MoebiusTransformation| if both factors are |MoebiusTransformation|s.
+
+    Parameters
+    ----------
+    coefficients
+        A tuple, list or |NumPy array| containing the four coefficients `a,b,c,d`.
+    normalize
+        If `True`, the coefficients are normalized, i.e. :math:`ad-bc=1`. Defaults to `False`.
+    name
+        Name of the transformation.
+    """
+
+    def __init__(self, coefficients, normalize=False, name=None):
+        assert len(coefficients) == 4
+        coefficients = np.array(coefficients, dtype=complex)
```

Comment:
What would be the reason to convert to `complex`? (Note that NumPy has `np.complex64`, `np.complex128`, and `np.complex256`.)

Reply:
This is not needed, you are right. I was afraid that the old Python behaviour where int divisions return ints not floats could lead to problems in `__call__`... 👻

Reply:
This is actually needed to resolve the `TypeError` that results from `/=`. Not sure exactly why though...